Decisions/scs-XXXX-v1-requirements-for-container-registry.md

```diff
@@ -0,0 +1,292 @@
+      It powers Red Hat enterprise products Red Hat Quay and Quay.io, which are used in
+      a productive way by many. Therefore, the project's maturity is at the good level
+  - Security
+    - The security disclosure is handled via a dedicated email address
+    - The code is reviewed within a standard PR process
+  - Activity
+    - 50+ contributors, 200+ forks, 2k+ GitHub stars
+    - The project has been used by many [companies](https://www.projectquay.io) that
+      run Quay in their production environments
+  - Lock-in risk assessment
+    - The project's owners/maintainers list is not publicly available and is stored in
+      the [downstream repository](https://github.com/quay/quay-docs#how-do-i-set-up). 
+      Therefore, it is hard to distinguish the risk of project failure caused by low
+      diversity across the companies. This should be improved
+
+- :x: [Keppel](https://github.com/sapcc/keppel)
+  - The project seems to be not widely used in a productive way and also the activity
+    around is currently not on a good level (5+ contributors). The development 
+    process as well as the design process seem to be open, but not documented yet
+
+- :x: [Nexus](https://github.com/sonatype/nexus-public)
+  - Nexus is an **open core** software that offers paid [pro version](https://www.sonatype.com/products/repository-oss-vs-pro-features) with advanced features
+
+- :x: [JFrog](https://jfrog.com/community/open-source/)
+  - JFrog Artifactory is shipped as an **open core** [software](https://jfrog.com/community/open-source/)
+    with limited features. The software is primarily offered as a paid [pro version](https://jfrog.com/pricing/#devops-onprem)
+
+- :x: [Kraken](https://github.com/uber/kraken)
+  - It seems that the project is not actively maintained as is discussed in the related
+    project's [issue](https://github.com/uber/kraken/issues/313) 
+
+- :x: [Portus](https://github.com/SUSE/Portus)
+  - It seems that the project is not actively maintained as is discussed in the related
+    project's [issue](https://github.com/SUSE/Portus/issues/2352) 
+
+## Required and desirable features check
+
+This section does an overview of the feature set of open source container registry 
+implementations (which passed the OSS health stage above) and map out requirements 
+(and nice-to-haves) against it. SCS-compliant software must be robust enough, to be able 
+to operate under heavy load (e.g. high availability (HA) mode, federation, etc.) and 
+the crucial feature is security.
+We defined a set of required features that the container registry implementation must 
+have and also a set of desirable (nice to have) features are defined and evaluated here. 
+
+
+**Required features**
+
+- Audit Logs
+  - Ability to record use in auditable logs so that activity can be traced to a single user
+- Authentication
```

Comment:
```suggestion
- Authentication of system identities
  - Support for authenticating system identities like Kubernetes clusters. Ideally supporting dynamic identity tokens from some IdP; Less ideal: Supporting static "system tokens"
- Authentication of users
```

Setting up authentication between e. g. Kubernetes clusters (or Pods) and a container registry is a big part of the overall user experience of the registry.

Overall, there are two general options that I know of:

1. The user fetches some (static) access token from the container registry, puts it into a K8s `Secret`, references such secret in the `imagePullSecrets` section of a Pod.
  - Requires management of a static token: Quite cumbersome, non-ideal security
2. The user grants the system identity of the Kubernetes nodes the permission to access the registry
  - Authentication is done out-of-sight, using IdP policies: Pretty much ideal

(Granted, the second option AFAIK is not as widely used as the first one, and as such, will require more work.)

I think the whole topic is relevant enough to be mentioned explicitly, while striving to implement the most ideal solution.